### PR TITLE
[opt](inverted index) topn opt reads only limit number of records

### PR DESCRIPTION
**Pull request by zzzxl1993**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/olap/iterators.h [hunk 1]
```diff
@@ -119,6 +119,7 @@ class StorageReadOptions {
     // slots that cast may be eliminated in storage layer
     std::map<std::string, PrimitiveType> target_cast_type_for_variants;
     RowRanges row_ranges;
+    size_t _topn_limit = 0;
```
Comment (xiaokang) [inline be/src/olap/iterators.h:122]: use consistent name style topn_limit

File: be/src/olap/rowset/rowset_reader_context.h [hunk 2]
```diff
@@ -84,6 +84,7 @@ struct RowsetReaderContext {
     // slots that cast may be eliminated in storage layer
     std::map<std::string, PrimitiveType> target_cast_type_for_variants;
     int64_t ttl_seconds = 0;
+    size_t _topn_limit = 0;
```
Comment (xiaokang) [inline be/src/olap/rowset/rowset_reader_context.h:87]: topn_limit

**Review by airborne12 — APPROVED** [review]
LGTM

**Comment by zzzxl1993** [comment]
run buildall

**Review by xiaokang — APPROVED** [review]
LGTM

**Pull request merged**